src/UriBuilder.Fluent/TerribleDevUriExtensions.cs

```diff
@@ -17,21 +17,25 @@ public static class TerribleDevUriExtensions
         /// <returns></returns>
         public static UriBuilder WithParameter(this UriBuilder bld, string key, params string[] values) => bld.WithParameter(key, valuesEnum: values);
 
+
         /// <summary>
-        /// Appends query strings from dictionary
+        /// Appends query strings from a list of key-value pairs (usually a dictionary).
         /// </summary>
         /// <param name="bld"></param>
         /// <param name="parameterDictionary"></param>
         /// <exception cref="ArgumentNullException"></exception>
         /// <returns></returns>
-        public static UriBuilder WithParameter(this UriBuilder bld, IDictionary<string, string> parameterDictionary)
+        public static UriBuilder WithParameter(this UriBuilder bld, IEnumerable<KeyValuePair<string, string>> parameterDictionary)
```

Comment:
As far as a dll is concerned this is actually a breaking change since the method signature is different (even though its pretty much the same thing. We should probably take the OG method as well, it just effects code that has already been compiled against the previous DLL. The consumer dll will have to be rebuilt 

![image](https://github.com/TerribleDev/UriBuilder.Fluent/assets/3459800/b03e83be-5d52-4582-9ef7-88af7f6a27bf)

Reply:
@TonyValenti it's fine, I'll cut a breaking change anyway since we're retargeting the RIDs

Reply:
I know its a different signature, but I know IDictionary inherits from IEnumerable<string,string> so atleast if people recompile it won't be an issue

Reply:
Hi @TerribleDev - My thoughts about the break:
It is a major version update so we're OK.
It is a binary break, not a source break so very minor.
Also, having both can cause ambiguities that arent fun to deal with.

Are you good with leaving things as I implemented?